mantidproject/mantid
# Add plotDOS function to directtools

This PR adds the capability to plot density of states to the `directtools` Python module. The capability comes in the form of a `plotDOS` functon.

**Report to:** [nobody].

**To test:**

With unit test data in Mantid's data search directories, play around with the following script:

```python
import directtools as dt

DirectILLCollectData(Run='ILL/IN4/084447.nxs', OutputWorkspace='data')
DirectILLReduction(InputWorkspace='data', OutputWorkspace='SofQW', OutputSofThetaEnergyWorkspace='SofTW')
dosFromStw = ComputeIncoherentDOS('SofTW')
dosFromSqw = ComputeIncoherentDOS('SofQW')
figure, axes = dt.plotDOS([dosFromStw, dosFromSqw], labels=[r'from $S(2\theta, E)$', r'from $S(Q,E)$'])
figure.show()
```

You can try:
- Use workspace names in ADS instead of the Python variables
- Try plotting a single workspace without the `list`
- Test the other arguments, for example `style='m', xscale='log'`, yscale='linear'`
- See what happens if you try to use the function to plot something else, for example DOS from another instrument
- Try to give some bogus inputs

Does everything work as expected? In case of errors, are the messages clear?

Build the `docs-html` target and check the documentation in Techniques->DirecttoolsPythonModule. Is the example useful? How about the functions reference documentation? Are there typos?

Fixes #22202.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
